### DISCUSSION TITLE
Tags: difference between poll and Poll

### DISCUSSION THREAD

**Discussion by LFrank2021** [opendcim/openDCIM#1561] [General]
I just stumbled over the case sensitivity of tags.
I accidentally typed 'poll' the first time for one of my devices and it would not show port states.
Even retyping it as 'Poll' would result in 'poll'.
Typing in both values resulted in:
```
PHP Fatal error:  Uncaught PDOException: SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '986-5' for key 'PRIMARY' in /srv/openDCIM/openDCIM-23.04/classes/Device.class.php:1856\nStack trace:\n#0 /srv/openDCIM/openDCIM-23.04/classes/Device.class.php(1856): PDO->exec()\n#1 /srv/openDCIM/openDCIM-23.04/devices.php(719): Device->SetTags()\n#2 {main}\n  thrown in /srv/openDCIM/openDCIM-23.04/classes/Device.class.php on line 1856,
```
I then resorted to manually updating the table:
```
update fac_Tags set Name="Poll" where TagID = 5;
``` 
Now the states of my connected ports are shown correctly.

Is this intentional or a flaw?

Kind regards
Frank

**Comment by wilpig** [comment]
That's going to take some thinking.  It's unintentional but it's going to be partly on how the sql server is making the keypair.  We didn't intentially make that all upper or lowercase so it is doing that on its own but the tags were supposed to be whatever case the user set.

**Reply by wilpig** [reply]
I will make a note and try to recreate this on dev.opendcim.org or if you can recreate it there and tell me which device you've done it to it will save me some time in trying to debug.

**Reply by LFrank2021** [reply]
I set 'poll' on https://dev.opendcim.org/devices.php?DeviceID=4312
Had to add a hdd_count too.
Then tried to added 'Poll', which resulted in white page and error entry.
Only 'Poll' will work with state polling. Easiest could be to allow Poll and poll for polling of devices.
Btw for me it is a CDU device.

**Reply by wilpig** [reply]
Excellent.  I'll try to look at that this afternoon when I am finished with $dayJob
